# Урок 6. Взаимосвязь величин. Параметрические и непараметрические показатели корреляции. Корреляционный анализ.

In [76]:
import numpy as np
import pandas as pd

1. Даны значения величины заработной платы заемщиков банка (zp) и значения их поведенческого кредитного скоринга (ks):
zp = [35, 45, 190, 200, 40, 70, 54, 150, 120, 110],
ks = [401, 574, 874, 919, 459, 739, 653, 902, 746, 832].

In [77]:
zp = np.array([35, 45, 190, 200, 40, 70, 54, 150, 120, 110])
ks = np.array([401, 574, 874, 919, 459, 739, 653, 902, 746, 832])

Найдите ковариацию этих двух величин с помощью элементарных действий, а затем с помощью функции cov из numpy.
Полученные значения должны быть равны.

In [78]:
np.mean(zp*ks) - np.mean(zp)*np.mean(ks)

9157.839999999997

In [79]:
np.cov(zp, ks, ddof=0)[0][1]

9157.84

Найдите коэффициент корреляции Пирсона с помощью ковариации и среднеквадратичных отклонений двух признаков,
а затем с использованием функций из библиотек numpy и pandas.

In [80]:
(np.mean(zp*ks) - np.mean(zp)*np.mean(ks))/(np.std(zp) * np.std(ks))

0.8874900920739158

In [81]:
(np.cov(zp, ks)/(np.std(zp, ddof=1)*np.std(ks, ddof=1)))[0][1]

0.887490092073916

In [82]:
np.corrcoef(zp, ks)[0][1]

0.8874900920739162

In [83]:
zp_pd = pd.Series(zp)
ks_pd = pd.Series(ks)
zp_pd.corr(ks_pd)

0.8874900920739162

2. Измерены значения IQ выборки студентов,
обучающихся в местных технических вузах:
131, 125, 115, 122, 131, 115, 107, 99, 125, 111.
Известно, что в генеральной совокупности IQ распределен нормально.
Найдите доверительный интервал для математического ожидания с надежностью 0.95.

In [84]:
iq = np.array([131, 125, 115, 122, 131, 115, 107, 99, 125, 111])

В данном задании нам неизвестно среднеквадратическое отклонение генеральной совокупности, значит воспользуемся распределением Стьюдента для определения доверительного интервала. Также потребуется несмещенная оценка среднеквадратического отклонения.

$$t_{табл} = 2.262$$

In [85]:
sigma = np.std(iq, ddof=1)
sigma

10.54566788359614

In [86]:
iq_mean = np.mean(iq)
iq_mean

118.1

Тогда,
### $$\mu_{IQ} \in [118.1 - 2.262\cdot\frac{10.5457}{\sqrt{10}};118.1 + 2.262\cdot\frac{10.5457}{\sqrt{10}}] \Rightarrow \mu_{IQ}  \in [110.5566; 125.6434]$$

In [87]:
print(iq_mean-sigma*2.262/np.sqrt(10))
print(iq_mean+sigma*2.262/np.sqrt(10))

110.55660776308164
125.64339223691834


3. Известно, что рост футболистов в сборной распределен нормально
с дисперсией генеральной совокупности, равной 25 кв.см. Объем выборки равен 27,
среднее выборочное составляет 174.2. Найдите доверительный интервал для математического ожидания с надежностью 0.95.

В данном случае известно среднеквадратическое отклонение генеральной совокупности, значит необходимо воспользоваться таблицей стандартного нормального распределения. Из этой таблицы можно получить информацию, что левее значения $z=1.96$ находится $\frac{\alpha}{2} = \frac{1-0.95}{2} = 2.5\%$ совокупности. Тогда доверительный интервал будет выглядеть следующим образом:
### $$\mu \in [174.2-1.96\cdot\frac{\sqrt{25}}{\sqrt{27}};174.2+1.96\cdot\frac{\sqrt{25}}{\sqrt{27}}] \Rightarrow \mu \in [172.3140;176.0860]$$

In [88]:
print(174.2 - 1.96*np.sqrt(25)/np.sqrt(27))
print(174.2 + 1.96*np.sqrt(25)/np.sqrt(27))

172.31398912064722
176.08601087935276
